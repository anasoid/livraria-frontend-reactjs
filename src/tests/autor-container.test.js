import React from 'react';
import renderer from 'react-test-renderer';
import AutorConteiner from '../components/autor/autor-container';
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import initialState from './initial-state';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe('test AutorConteiner component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <AutorConteiner />
      </Provider>
    );
  });

  it('should renders without crashing', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <AutorConteiner />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('AutorConteiner should be contain form', () => {
    expect(wrapper.find('form').at(0)).toExist();
  });

  it('should be contains a "Novo Autor" title', () => {
    expect(wrapper.find('h1').at(0)).toHaveText('Novo Autor');
  });

  it('should be contains nome input', () => {
    const nome = wrapper.find('#autorNome').at(0);
    expect(nome).toMatchSelector('input');
    expect(nome).toHaveProp('name', 'nome');
  });

  it('should be contains email input', () => {
    const email = wrapper.find('#autorEmail').at(0);
    expect(email).toMatchSelector('input');
    expect(email).toHaveProp('name', 'email');
  });

  it('should be contain a button "Gravar Autor" and type submit', () => {
    const button = wrapper.find('#btnGravarAutor').at(0);
    expect(button).toHaveProp('type', 'submit');
    expect(button).toHaveText('Gravar Autor');
  });

});