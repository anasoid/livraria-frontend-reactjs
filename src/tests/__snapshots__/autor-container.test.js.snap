// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AutorConteiner component should renders without crashing 1`] = `
<div>
  <h1>
    Novo Autor
  </h1>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="autorNome"
      >
        Nome: 
      </label>
      <input
        id="autorNome"
        name="nome"
        onChange={[Function]}
        value=""
      />
      <br />
      <label
        htmlFor="autorEmail"
      >
        Email: 
      </label>
      <input
        id="autorEmail"
        name="email"
        onChange={[Function]}
        value=""
      />
      <br />
      <button
        disabled={true}
        id="btnGravarAutor"
        type="submit"
      >
        Gravar Autor
      </button>
    </form>
  </div>
  <br />
  <div>
    <table>
      <caption>
        Autores
      </caption>
      <thead>
        <tr>
          <th>
            Nome
          </th>
          <th>
            Email
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
